tools/getrealdeps.rb

```diff
@@ -1,121 +1,141 @@
+    end
+    abort("Package #{pkg} either does not exist or does not contain any libraries.") unless File.exist?(pkgfilelist)
+  end
 
-pkgfiles = crewfilesfxn(pkg).split("\n").uniq
+  # Speed up grep.
+  ENV['LC_ALL'] = 'C'
 
-FileUtils.rm_rf("/tmp/deps/#{pkg}")
-# Remove files we don't care about, such as man files and non-binaries.
-pkgfiles = pkgfiles.reject { |i| !File.file?(i.chomp) || !File.read(i.chomp, 4) == "\x7FELF" || i.include?('.zst') }
+  # Install grep if a functional local copy does not exist.
+  if system('grep --version > /dev/null 2>&1')
+    @grep = 'grep'
+  else
+    system('crew install grep')
+    @grep = "#{CREW_PREFIX}/bin/grep"
+  end
 
-# Use readelf to determine library dependencies, as
-# this doesn't almost run a program like using ldd would.
-pkgdepsfiles = pkgfiles.map do |i|
-  system("upx -d #{i} > /dev/null 2>&1")
-  FileUtils.mkdir_p("/tmp/deps/#{pkg}/")
-  `readelf -d "#{i}" 2>/dev/null | #{GREP} NEEDED | awk '{print $5}' | sed 's/\\[//g' | sed 's/\\]//g' | awk '!x[$0]++' | tee /tmp/deps/#{pkg}/#{File.basename(i)}`
-end
-pkgdepsfiles = pkgdepsfiles.map do |filedeps|
-  filedeps.split("\n")
-end.flatten.compact.uniq
+  # Gawk is needed for adding dependencies.
+  unless system('gawk -W version > /dev/null 2>&1')
+    puts "\nThe inplace replacement functionality of gawk is used to add missing dependencies to package files."
+    system('crew install gawk')
+  end
+
+  # upx is needed to expand compressed binaries to check for dependencies.
+  unless system('upx --version > /dev/null 2>&1')
+    puts "\nUpx is needed to expand compressed binaries."
+    system('crew install upx')
+  end
 
-# Figure out which Chromebrew packages provide the relevant deps.
-pkgdeps = pkgdepsfiles.map do |file|
-            whatprovidesfxn(file, pkg)
-          end.sort.reject { |i| i.include?(pkg) }.map { |i| i.split("\n")}.flatten.uniq
+  # What files does the package provide.
+  pkgfiles = File.read(pkgfilelist).split("\n").uniq
 
-# Massage the glibc entries in the dependency list.
-pkgdeps = pkgdeps.map { |i| i.gsub(/glibc_build.*/, 'glibc') }.uniq
-pkgdeps = pkgdeps.map { |i| i.gsub(/glibc_lib.*/, 'glibc_lib') }.uniq.map(&:strip).reject(&:empty?)
+  # Look at files in CREW_DEST_DIR instead of assuming the package is
+  # normally installed, which lets us avoid installing the package if it
+  # was just built.
+  pkgfiles.map! {|item| item.prepend(CREW_DEST_DIR)} if @opt_use_crew_dest_dir
 
-# Look for missing runtime dependencies.
-missingpkgdeps = pkgdeps.reject { |i| File.read("#{CREW_PREFIX}/lib/crew/packages/#{pkg}.rb").include?("depends_on '#{i}'") unless File.read("#{CREW_PREFIX}/lib/crew/packages/#{pkg}.rb").include?("depends_on '#{i}' => :build") }
+  FileUtils.rm_rf("/tmp/deps/#{pkg}")
+  # Remove files we don't care about, such as man files and non-binaries.
+  pkgfiles = pkgfiles.reject { |i| !File.file?(i.chomp) || !File.read(i.chomp, 4) == "\x7FELF" || i.include?('.zst') }
```

Comment:
Should the compression algorithm be abstracted with a variable?

Reply:
The zst files are from man pages... otherwise it would be a gz file... but in any case, readelf will reject the files anyways.

Reply:
So I don't think it makes sense to make it any more complicated. This is a 99.9% solution for the usage of this tool, and the other uses would be for checking existing packages ~ 2 years old or older...

Reply:
This entire PR is full of complexities. :sweat: